### PR TITLE
[FIX] account,*: multiple fixes related to UBL, Peppol and Hungary

### PR DESCRIPTION
**Pull request by clbr-odoo**
### [FIX] account_edi_ubl_cii,l10n_ro_edi: Allow NO VAT tax scheme ID
Rule [BR-CO-09] enforce the VAT number to start with a country code
if the tax scheme id is "VAT".
In some countries (RO, HU, non-EU countries, ...) it is perfectly valid
to have a VAT that doesn't follow this format.
To avoid the rule to raise, we should set the tax scheme ID to something
else than "VAT".

### [FIX] account_edi_ubl_cii: only set BIS3 format by default on restricted countries
BIS3 is an electronic format closely related to Peppol.
It therefore only make sense to enable it by default on partners
that are in countries where it's actively used or will become
mandatory soon.

### [FIX] account_edi_ubl_cii,base_vat,l10n_hu: VAT conversion to EU format
Hungary may have different format of VAT number.
It is possible to convert their local format (xxxxxxxx-y-zz) to the EU
format (HUxxxxxxxx).
We should use this converted format when computing the Peppol Endpoint.

Enterprise: https://github.com/odoo/enterprise/pull/75671

task-4374458

### CODE REVIEW

File: addons/l10n_hu/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    l10n_hu_eu_vat = fields.Char(compute='_compute_l10n_hu_eu_vat')
+
+    def _compute_l10n_hu_eu_vat(self):
+        hu_vat_partners = self.filtered(lambda p: p.country_code == 'HU' and p.vat)
+        for partner in hu_vat_partners:
+            partner.l10n_hu_eu_vat = partner._convert_hu_local_to_eu_vat(partner.vat)
+        (self - hu_vat_partners).l10n_hu_eu_vat = False
```
Comment (smetl) [inline addons/l10n_hu/models/res_partner.py:13]:
```suggestion
        hu_vat_partners = self.filtered(lambda p: p.country_code == 'HU' and p.vat)
        for partner in self:
            if partner.country_code == 'HU' and partner.vat:
                partner.l10n_hu_eu_vat = partner._convert_hu_local_to_eu_vat(partner.vat)
            else:
                partner.l10n_hu_eu_vat = False
```

Comment (smetl) [reply addons/l10n_hu/models/res_partner.py:13]: No dependency?

**Review by smetl — APPROVED** [review]

**Comment by Xavier-Do** [comment]
Upgrade exception [#460](https://runbot.odoo.com/odoo/runbot.upgrade.exception/460?menu_id=405) added.

Waiting the forward-port of this pr, the exception will be applied on all builds, and create an inconsistent state for migrations.
**Please forward port this asap up to master without change.** Exception should be forward-ported before the end of the week.

PRs adding fields/models should be merged at the beginning of the week and all the forward ports should be merged in the same week.

If you need to apply any change before it reaches master, please notify runbot team.

Details:
```
field:res.users.l10n_hu_eu_vat
field:res.partner.l10n_hu_eu_vat
```
cc @KangOl @nseinlet @aj-fuentes

**Comment by smetl** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**